lwt/happy_eyeballs_lwt.ml

```diff
@@ -10,16 +10,21 @@ module Log = (val Logs.src_log src : Logs.LOG)
 
 let now = Mtime_clock.elapsed_ns
 
+type getaddrinfo = [ `A | `AAAA ] -> [ `host ] Domain_name.t -> (Ipaddr.Set.t, [ `Msg of string ]) result Lwt.t
+
 type t = {
   mutable waiters : ((Ipaddr.t * int) * Lwt_unix.file_descr, [ `Msg of string ]) result Lwt.u Happy_eyeballs.Waiter_map.t ;
   mutable cancel_connecting : (int * unit Lwt.u) list Happy_eyeballs.Waiter_map.t;
   mutable he : Happy_eyeballs.t ;
-  dns : Dns_client_lwt.t ;
   timer_interval : float ;
   timer_condition : unit Lwt_condition.t ;
   counter : int ;
+  mutable getaddrinfo : getaddrinfo ;
 }
 
+let inject getaddrinfo t =
+  t.getaddrinfo <- getaddrinfo
```

Comment:
I'd appreciate a log (warning?) if there's a second injection... i.e. if `getaddrinfo` != stub, explain to the user that there's something being overwritten.

Reply:
and indeed, now that we revised how to do things, I'd be happy to remove the inject... and have `getaddrinfo` not mutable (in case this is fine with the dns-client-lwt usecase)?